tests/integration/test_proxy.py

```diff
@@ -59,3 +89,23 @@ def test_use_proxy(tmpdir, httpbin, proxy_server):
         assert cassette_response.headers[key] == response.headers[key]
     assert cassette_response.headers == response.headers
     assert cassette.play_count == 1
+
+
+def test_use_https_proxy(tmpdir, httpbin_secure, proxy_server):
+    """Ensure that it works with an HTTPS proxy."""
+    with vcr.use_cassette(str(tmpdir.join("proxy.yaml"))):
+        response = requests.get(httpbin_secure.url, proxies={"https": proxy_server})
+
+    with vcr.use_cassette(str(tmpdir.join("proxy.yaml")), mode="once") as cassette:
+        cassette_response = requests.get(
+            httpbin_secure.url,
+            proxies={"https": proxy_server},
+        )
+
+    for key in set(cassette_response.headers.keys()) & set(response.headers.keys()):
+        assert cassette_response.headers[key] == response.headers[key]
+    assert cassette_response.headers == response.headers
+    assert cassette.play_count == 1
+
+    # The cassette url points to httpbin, not the proxy
```

Comment:
```suggestion
    # The cassette URL points to httpbin, not the proxy
```
(not critical, true)